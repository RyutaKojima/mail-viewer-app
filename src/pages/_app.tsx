import { AppProps } from 'next/app'
import { ChakraProvider } from '@chakra-ui/react'
import { Analytics } from '@vercel/analytics/react'

export default function MyApp({ Component, pageProps }: AppProps) {
  return (
    <ChakraProvider>
      <Component {...pageProps} />
      <Analytics />
    </ChakraProvider>
  )
}
