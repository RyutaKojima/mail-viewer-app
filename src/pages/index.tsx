import Head from 'next/head'
import { Box, Center, chakra, Grid, Textarea } from '@chakra-ui/react'
import { EmailIcon } from '@chakra-ui/icons'
import React from 'react'
import dynamic from 'next/dynamic'

const MailParserView = dynamic(() => import('../components/MailParserView'), {
  ssr: false,
})

export const Index = (): JSX.Element => {
  const [value, setValue] = React.useState('')

  const handleInputChange = (e) => {
    const inputValue = e.target.value
    setValue(inputValue)
  }

  return (
    <Box bg="gray.700" color="gray.200" p="2" h="100%" minH="100vh">
      <Head>
        <title>Mail Viewer</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <chakra.header p="2">
        <Center>
          <chakra.h1>
            <EmailIcon />
            Mail Viewer
          </chakra.h1>
        </Center>
      </chakra.header>

      <main>
        <Grid templateColumns="repeat(2, 1fr)" gap={6}>
          <Box
            w="100%"
            minH="200px"
            p={2}
            borderWidth="1px"
            borderRadius="lg"
            overflow="hidden"
          >
            <Textarea
              value={value}
              onChange={handleInputChange}
              placeholder="ここにメールデータを入力"
              w="100%"
              h="100%"
              border="none"
            />
          </Box>

          <Box
            w="100%"
            minH="200px"
            p={2}
            borderWidth="1px"
            borderRadius="lg"
            overflow="hidden"
          >
            <MailParserView mail={value} />
          </Box>
        </Grid>
      </main>

      <chakra.footer mt="4" p="2">
        <Center>
          <a
            href="https://github.com/RyutaKojima/mail-viewer-app"
            target="_blank"
            rel="noopener noreferrer"
          >
            Made by RyutaKojima
          </a>
        </Center>
      </chakra.footer>
    </Box>
  )
}

export default Index
