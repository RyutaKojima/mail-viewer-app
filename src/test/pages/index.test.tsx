import React from 'react'
import { render } from '../testUtils'
import { Index } from '../../pages'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Index />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
