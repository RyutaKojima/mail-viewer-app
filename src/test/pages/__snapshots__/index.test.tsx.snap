// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1tyio8u"
  >
    <header
      class="css-1o3848f"
    >
      <div
        class="css-gmuwbf"
      >
        <h1
          class="css-0"
        >
          <svg
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              fill="currentColor"
            >
              <path
                d="M11.114,14.556a1.252,1.252,0,0,0,1.768,0L22.568,4.87a.5.5,0,0,0-.281-.849A1.966,1.966,0,0,0,22,4H2a1.966,1.966,0,0,0-.289.021.5.5,0,0,0-.281.849Z"
              />
              <path
                d="M23.888,5.832a.182.182,0,0,0-.2.039l-6.2,6.2a.251.251,0,0,0,0,.354l5.043,5.043a.75.75,0,1,1-1.06,1.061l-5.043-5.043a.25.25,0,0,0-.354,0l-2.129,2.129a2.75,2.75,0,0,1-3.888,0L7.926,13.488a.251.251,0,0,0-.354,0L2.529,18.531a.75.75,0,0,1-1.06-1.061l5.043-5.043a.251.251,0,0,0,0-.354l-6.2-6.2a.18.18,0,0,0-.2-.039A.182.182,0,0,0,0,6V18a2,2,0,0,0,2,2H22a2,2,0,0,0,2-2V6A.181.181,0,0,0,23.888,5.832Z"
              />
            </g>
          </svg>
          Mail Viewer
        </h1>
      </div>
    </header>
    <main>
      <div
        class="css-1p6s0yy"
      >
        <div
          class="css-da66gq"
        >
          <textarea
            class="chakra-textarea css-11663sq"
            placeholder="ここにメールデータを入力"
          />
        </div>
        <div
          class="css-da66gq"
        />
      </div>
    </main>
    <footer
      class="css-1f7brxj"
    >
      <div
        class="css-gmuwbf"
      >
        <a
          href="https://github.com/RyutaKojima/mail-viewer-app"
          rel="noopener noreferrer"
          target="_blank"
        >
          Made by RyutaKojima
        </a>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
